DSharpPlus/Net/Gateway/Compression/Zlib/ZlibPayloadDecompressor.cs

```diff
@@ -0,0 +1,46 @@
+#pragma warning disable IDE0046
+
+using System;
+
+using CommunityToolkit.HighPerformance;
+using CommunityToolkit.HighPerformance.Buffers;
+
+namespace DSharpPlus.Net.Gateway.Compression.Zlib;
+
+/// <summary>
+/// A payload decompressor using zlib on the payload level.
+/// </summary>
+public sealed class ZlibPayloadDecompressor : IPayloadDecompressor
+{
+    /// <inheritdoc/>
+    public string? Name => null;
+
+    /// <inheritdoc/>
+    public bool IsTransportCompression => false;
+
+    /// <inheritdoc/>
+    public bool TryDecompress(ReadOnlySpan<byte> compressed, ArrayPoolBufferWriter<byte> decompressed)
+    {
+        using ZlibWrapper wrapper = new();
+
+        if (!wrapper.TryInflate(compressed, decompressed))
+        {
+            decompressed.Clear();
```

Comment:
Do we need to clear here?

Reply:
yes, it may have been partially decompressed and then encountered an error (though extremely unlikely, clearing here will make the error more obvious)